Common/Data/Slice.cs

```diff
@@ -444,6 +446,50 @@ public override bool TryGetValue(Symbol symbol, out dynamic data)
             return false;
         }
 
+        /// <summary>
+        /// Merge two slice with same Time
+        /// </summary>
+        /// <param name="inputSlice">slice instance</param>
+        public void MergeSlice(Slice inputSlice)
+        {
+            if (Time != inputSlice.Time)
+            {
+                throw new Exception($"Slice with time {Time} can't be merged with given slice with different {inputSlice.Time}");
+            }
+
+            _bars = (TradeBars)UpdateCollection<TradeBar>(_bars, inputSlice.Bars);
+            _quoteBars = (QuoteBars)UpdateCollection<QuoteBar>(_quoteBars, inputSlice.QuoteBars);
+            _ticks = (Ticks)UpdateCollection<List<Tick>>(_ticks, inputSlice.Ticks);
+            _optionChains = (OptionChains)UpdateCollection<OptionChain>(_optionChains, inputSlice.OptionChains);
+            _futuresChains = (FuturesChains)UpdateCollection<FuturesChain>(_futuresChains, inputSlice.FuturesChains);
+            _splits = (Splits)UpdateCollection<Split>(_splits, inputSlice.Splits);
+            _dividends = (Dividends)UpdateCollection<Dividend>(_dividends, inputSlice.Dividends);
+            _delistings = (Delistings)UpdateCollection<Delisting>(_delistings, inputSlice.Delistings);
+            _symbolChangedEvents = (SymbolChangedEvents)UpdateCollection<SymbolChangedEvent>(_symbolChangedEvents, inputSlice.SymbolChangedEvents);
+
+            // Should keep this._rawDataList last so that selected data points are not overriden
+            // while creating _data
+            var tempRawDataList = inputSlice._rawDataList;
+            tempRawDataList.AddRange(_rawDataList);
+            _rawDataList = tempRawDataList; 
```

Comment:
Similar to `UpdateCollection<T>` can check counts to be super lazy about the `AddRange` call 👍